actions/service.go

```diff
@@ -5,94 +5,113 @@ package actions
 import (
 	"context"
 	"errors"
-	"strings"
+	"fmt"
 	"time"
 
-	"github.com/google/uuid"
 	"github.com/hashicorp/go-multierror"
 	"github.com/treeverse/lakefs/db"
+	"github.com/treeverse/lakefs/graveler"
 )
 
 type Service struct {
-	DB db.Database
+	DB     db.Database
+	Source Source
+	Writer OutputWriter
 }
 
 type Task struct {
-	RunID  string
-	Action *Action
-	HookID string
-	Hook   Hook
+	RunID     string
+	HookRunID string
+	Action    *Action
+	HookID    string
+	Hook      Hook
+	Err       error
+	StartTime time.Time
+	EndTime   time.Time
+	NextTask  *Task
+}
+
+type RunResult struct {
+	RunID     string    `db:"run_id"`
+	BranchID  string    `db:"branch_id"`
+	SourceRef string    `db:"source_ref"`
+	EventType string    `db:"event_type"`
+	StartTime time.Time `db:"start_time"`
+	EndTime   time.Time `db:"end_time"`
+	Passed    bool      `db:"passed"`
+	CommitID  string    `db:"commit_id"`
```

Comment:
Nullable, no?

Reply:
I prefer to have an empty string as no commit_id instead of playing with nil